tools/nbs_decoder/__init__.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+
+import xxhash
+import gzip
+import struct
+import sys
+import os
+import pkgutil
+import glob
+import enum
+import b
+import google.protobuf.message
+from google.protobuf.json_format import MessageToJson
+
+# Build all the protocol buffers into python messages located at out_folder
+protobuf_path = os.path.join(os.path.dirname(__file__), "protocol_buffers")
+os.makedirs(protobuf_path, exist_ok=True)
+
+# Find protobufs to build
+nuclear_proto_dir = os.path.join(b.nuclear_dir, "message", "proto")
+user_proto_dir = os.path.join(b.project_dir, "shared")
+
+proto_files = [
+    *glob.glob(os.path.join(nuclear_proto_dir, "**", "*.proto"), recursive=True),
+    *glob.glob(os.path.join(user_proto_dir, "**", "*.proto"), recursive=True),
+]
+
+# Build the protocol buffers
+if (
+    os.system(
+        "protoc --python_out={protobuf_path} -I{nuclear_dir} -I{user_dir} {proto_files}".format(
```

Comment:
We should do a version check on protoc to ensure it meets a minimum version requirement (if we have one)

Reply:
That's what the if is for here. If protoc can't parse it then it'll return an error code which means we can't use it. The shell is not redirected so you'll see whatever protoc complained about in your terminal.

Reply:
Its not a question of whether or not protoc will error on a "bad" file, but whether the error protoc will give will be ambiguous in the context

Reply:
I feel like it's just an extra check that's not really needed for a simple tool